GoRogue/MapGeneration/Region.cs

```diff
@@ -1,355 +1,478 @@
+            => InnerFromOuterPoints(new Area(outer));
+
         /// <summary>
-        /// Removes (from this region) the inner points shared in common
-        /// with the imposing region
+        /// Gets the points contains within a hollow area
         /// </summary>
-        /// <param name="imposing">the region to check for common outer points</param>
-        public void RemoveOverlappingInnerPoints(Region imposing)
+        /// <param name="outer">The outer area</param>
+        /// <returns></returns>
+        public static Area InnerFromOuterPoints(Area outer)
         {
-            foreach (Point c in imposing._innerPoints)
+            var outerList = outer.OrderBy(x => x.X).ToList();
+
+            if(outerList.Count == 0)
+                return new Area();
+
+            Area points = new Area();
+
+            for (int i = outerList[0].X + 1; i < outerList[^1].X; i++)
             {
-                while (_outerPoints.Contains(c))
-                    _outerPoints.Remove(c);
-                while (_innerPoints.Contains(c))
-                    _innerPoints.Remove(c);
+                List<Point> row = outerList.Where(point => point.X == i).OrderBy(point => point.Y).ToList();
+                if(row.Count > 0)
+                {
+                    for (int j = row[0].Y; j <= row[^1].Y; j++)
+                    {
+                        points.Add(new Point(i, j));
+                    }
+                }
             }
+
+            return points;
         }
+
+        /// <summary>
+        /// Returns a new Region from a parameter set similar to GoRogue.Rectangle
+        /// </summary>
+        /// <param name="origin"></param>
+        /// <param name="width"></param>
+        /// <param name="height"></param>
+        /// <returns>A new region in the shape of a rectangle</returns>
+        public static Region Rectangle(Point origin, int width, int height)
+            => FromRectangle(new Rectangle(origin, origin + (width, height)));
+
+        /// <summary>
+        /// Creates a new Region from a GoRogue.Rectangle.
+        /// </summary>
+        /// <param name="r">The rectangle</param>
+        /// <returns>A new region in the shape of a rectangle</returns>
+        public static Region FromRectangle(Rectangle r)
+            => new Region(r.MinExtent, (r.MaxExtentX, r.MinExtentY), r.MaxExtent, (r.MinExtentX, r.MaxExtentY));
+
+
+         /// <summary>
+         /// Creates a new Region in the shape of a parallelogram, with diagonals going down and right.
+         /// </summary>
+         /// <param name="origin">Origin of the parallelogram.</param>
+         /// <param name="width">Width of the parallelogram.</param>
+         /// <param name="height">Height of the parallelogram.</param>
+         /// <returns>A new region in the shape of a parallelogram</returns>
+         public static Region ParallelogramFromTopCorner(Point origin, int width, int height)
```

Comment:
Haven't tested, but does this function need to account for `Direction.YIncreasesUpward` as well?

Reply:
Yes, it does. Change incoming.